src/SelfUpdateCommand.php

```diff
@@ -75,26 +76,27 @@ protected function configure(): void
     /**
      * Get all releases from GitHub.
      *
-     * @throws \Exception
-     *
      * @return array
+     * @throws \Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface
+     *
+     * @throws \Exception
      */
     protected function getReleasesFromGithub(): array
     {
         $version_parser = new VersionParser();
+
         $opts = [
-            'http' => [
-                'method' => 'GET',
-                'header' => [
-                    'User-Agent: ' . $this->applicationName  . ' (' . $this->gitHubRepository . ')' . ' Self-Update (PHP)',
-                ],
+            'headers' => [
+                'User-Agent' => $this->applicationName  . ' (' . $this->gitHubRepository . ')' . ' Self-Update (PHP)',
             ],
         ];
+        $client = HttpClient::create($opts);
+        $response = $client->request(
+            'GET',
+            'https://api.github.com/repos/' . $this->gitHubRepository . '/releases'
+        );
 
-        $context = stream_context_create($opts);
-
-        $releases = file_get_contents('https://api.github.com/repos/' . $this->gitHubRepository . '/releases', false, $context);
-        $releases = json_decode($releases);
+        $releases = json_decode($response->getContent(), FALSE, 512, JSON_THROW_ON_ERROR);
```

Comment:
You could use `$response->toArray()`
ref: https://github.com/symfony/symfony/blob/99b25a1a9de655961ac7c8fbf18e5d00f84ed798/src/Symfony/Contracts/HttpClient/ResponseInterface.php#L61-L72

Reply:
Thanks, but the rest of the class assumes `$releases` is an object and I wanted to minimize refactoring for this PR